### PR TITLE
[notifications] Guard notifications from loading badgin in SSR

### LINKED ISSUE #10724: [nextjs] SSR failing due to dependency on `badgin` in `expo-notifications`
### Summary of Issue 
SSR in Expo + Next.js application failing due to `ReferenceError: window is not defined` which seems to be tied to importing the `badgin` library in `ExponentNotifications.web.js`.

### Environment - output of `expo diagnostics` & the platform(s) you're targeting
Web (with Next.js)
```
  Expo CLI 3.28.0 environment info:
    System:
      OS: macOS 11.0
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 12.18.1 - /usr/local/bin/node
      Yarn: 1.22.5 - /usr/local/bin/yarn
      npm: 6.14.5 - /usr/local/bin/npm
    SDKs:
      iOS SDK:
        Platforms: iOS 13.7, DriverKit 19.0, macOS 10.15, tvOS 13.4, watchOS 6.2
      Android SDK:
        API Levels: 23
        Build Tools: 23.0.1
        System Images: android-23 | Google APIs Intel x86 Atom
    IDEs:
      Xcode: 11.7/11E801a - /usr/bin/xcodebuild
    npmPackages:
      expo: ^39.0.0 => 39.0.3 
      react: 16.13.1 => 16.13.1 
      react-dom: 16.13.1 => 16.13.1 
      react-native: https://github.com/expo/react-native/archive/sdk-39.0.3.tar.gz => 0.63.2 
      react-native-web: ~0.13.7 => 0.13.18 
    npmGlobalPackages:
      expo-cli: 3.28.0
    Expo Workflow: managed
```

### PR DESCRIPTION
**Pull request by byCedric**
# Why

Fixes #10724

# How

This moves the `badgin` module loading to the method itself, to protect it on SSR.

# Test Plan

See this change in [this repo](https://github.com/byCedric/expo-issue-10724-workaround/commit/ff8ea03c2a8b8278304262042da4db82e97084bd) and try it out.

### CODE REVIEW

File: packages/expo-notifications/src/BadgeModule.web.ts [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
     return lastSetBadgeCount;
   },
-  setBadgeCountAsync: async (badgeCount: number, options?: badgin.Options) => {
+  setBadgeCountAsync: async (badgeCount, options) => {
+    // If this module is loaded in SSR (NextJS), we can't modify the badge.
+    // It also can't load the badgin module, that instantly invokes methods on window.
+    if (typeof window === 'undefined') {
+      return false;
+    }
+    const badgin = require('badgin');
```
Comment (sjchmiela) [inline packages/expo-notifications/src/BadgeModule.web.ts:17]: Maybe we could even `await import()`? 🤔 Would it work in SSR?

**Review by sjchmiela — APPROVED** [review]
Looks great! Thank you for taking care of it!

Would you mind updating `CHANGELOG` too? 🙏

File: packages/expo-notifications/src/BadgeModule.web.ts (hunk 1 above)
Comment (byCedric) [reply packages/expo-notifications/src/BadgeModule.web.ts:17]: I can try 😄

Comment (byCedric) [reply packages/expo-notifications/src/BadgeModule.web.ts:17]: I'm afraid not:

Screenshot 2020-10-20 at 14 26 04

File: packages/expo-notifications/CHANGELOG.md [hunk 2]
```diff
@@ -99,11 +100,7 @@ _This version does not introduce any user-facing changes._
...
...We are planning to deprecate the `.NOTIFICATIONS` permission soon.## 0.2.0 — 2020-05-27### 🛠 Breaking changes> Note that this may or may not be a breaking change for you — if you'd expect the notification to be automatically dismissed when tapped on this is a bug fix and a new feature (fixes inconsistency between platforms as on iOS this is the only supported behavior; adds the ability to customize the behavior on Android). If you'd expect the notification to only be dismissed at your will this is a breaking change and you'll need to add `autoDismiss: false` to your notification content inputs.- Changed the default notification behavior on Android to be automatically dismissed when clicked. This is customizable with the `autoDismiss` parameter of `NotificationContentInput`. ([#8241](https://github.com/expo/expo/pull/8241) by [@thorbenprimke](https://github.com/thorbenprimke))### 🎉 New features- Added the ability to configure whether the notification should be automatically dismissed when tapped on or not (on Android) with the `autoDismiss` parameter of `NotificationContentInput`. ([#8241](https://github.com/expo/expo/pull/8241) by [@thorbenprimke](https://github.com/thorbenprimke))
```
Comment (sjchmiela) [inline packages/expo-notifications/CHANGELOG.md:103]: Could you please revert these changes? (PR for fixing automatic addition of changelogs is awaiting merge)

Comment (byCedric) [reply packages/expo-notifications/CHANGELOG.md:103]: oh, I didn't even notice! I'll revert the change right away!

Comment (byCedric) [reply packages/expo-notifications/CHANGELOG.md]: Does this happen often? If so, maybe we should think about using ASTs to avoid parsing issues?

Comment (sjchmiela) [reply packages/expo-notifications/CHANGELOG.md]: We use `marked` to parse and then our own implementation of renderer in `expotools`. Yes, this does happen often on `expo-notifications` changelog as it's somehow complicated, but I think https://github.com/expo/expo/pull/10639 and https://github.com/expo/expo/pull/10659 should fix the problem at its source. 🙂

Comment (byCedric) [reply packages/expo-notifications/CHANGELOG.md]: Can we merge #10639 already? If so, I'll rebase on that 😄  That would solve the merge conflicts.

Comment (byCedric) [reply packages/expo-notifications/CHANGELOG.md]: nvm, I went for the full rollback and manual edit for my changelog entry.

**Review by cruzach — APPROVED** [review]
